pandas/core/internals/managers.py

```diff
@@ -338,6 +343,10 @@ def apply(
         aligned_args = {k: kwargs[k] for k in align_keys}
 
         for b in self.blocks:
+            if block_inplace:
+                # Check if inplace allowed according to CoW
+                if using_copy_on_write() and b.refs.has_reference():
```

Comment:
call using_copy_on_write once outside the loop?

Reply:
This shouldn't live here. These logic live on the block level or rather in ``operate_blockwise`` in this case.